spiped/main.c

```diff
@@ -292,12 +321,15 @@ main(int argc, char * argv[])
 
 	/* Start accepting connections. */
 	if ((dispatch_cookie = dispatch_accept(s, opt_t, opt_R ? 0.0 : opt_r,
-	    sas_t, opt_d, opt_f, opt_g, opt_j, K, opt_n, opt_o,
-	    opt_1 ? &conndone : NULL)) == NULL) {
+	    sas_t, opt_d, opt_f, opt_g, opt_j, K, opt_n, opt_o, opt_1,
+	    &conndone)) == NULL) {
 		warnp("Failed to initialize connection acceptor");
 		goto err5;
 	}
 
+	/* Check periodically whether a signal was received. */
+	events_timer_register_double(graceful_shutdown, dispatch_cookie, 1.0);
```

Comment:
This is not something I should change, but @gperciva might do after the
merge, right?
Am 14.05.2016 23:34 schrieb "Colin Percival" <notifications@github.com>:

> In spiped/main.c
> <https://github.com/Tarsnap/spiped/pull/51#discussion_r63282373>:
>
> >  		warnp("Failed to initialize connection acceptor");
> >  		goto err5;
> >  	}
> >
> > +	/* Check periodically whether a signal was received. */
> > +	events_timer_register_double(graceful_shutdown, dispatch_cookie, 1.0);
>
> @gperciva <https://github.com/gperciva> might want to record the cookie
> so that he can deregister the callback in the event of a clean exit so that
> it isn't flagged as a memory leak.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/Tarsnap/spiped/pull/51/files/b760ac699dd7f655055afe5840f27b44c578da81#r63282373>
>